deck-core/src/main/kotlin/com/deck/core/entity/impl/DeckRole.kt

```diff
@@ -0,0 +1,119 @@
+package com.deck.core.entity.impl
+
+import com.deck.common.entity.RawRole
+import com.deck.common.entity.RawRoleBotScope
+import com.deck.common.entity.RawRolePermissions
+import com.deck.common.entity.RawRolePermissionsOverwritten
+import com.deck.common.util.*
+import com.deck.core.DeckClient
+import com.deck.core.entity.Role
+import com.deck.core.entity.RoleBotScope
+import com.deck.core.entity.RolePermissions
+import com.deck.core.entity.RolePermissionsOverwritten
+
+public class DeckRole(
+    override val client: DeckClient,
+    public val raw: RawRole
```

Comment:
Same problem as before, no raw access, instead, provide all the necessary parameters on the constructor.